hack/testing-olm-upgrade/upgrade-common

```diff
@@ -397,15 +397,18 @@ discover_versions(){
   major_version="$(echo $full_version | cut -d'.' -f1)"
   minor_version="$(echo $full_version | cut -d'.' -f2)"
 
-  # we shouldn't run into this yet... maybe for 6.0 but we should like
-  # add the 'Replaces' spec field and use that for the previous_version
-  if [[ $minor_version -eq 0 ]]; then
-    log::info "Will be unable to calculate the previous_version since our minor version is 0"
-    exit 0
+  export version="$(echo $major_version.$minor_version)"
+
+  # Hard rewrite of 5.0.0 to superficial 4.7.0, as replacement for
+  # the breaking point when OpenShift Logging shifted release schema.
+  if [[ $major_version -eq 5 ]] && [[ $minor_version -eq 0 ]]; then
+    # Set this to 4.7 because we need to calculate the latest previous
+    # version prior to 5.0, which is 4.6 for OpenShift Logging.
+    major_version=4
+    minor_version=7
```

Comment:
previous version should be 4.7 or 4.6?

Reply:
Previous to 5.0 is 4.6, but in order to calculate that via `get_latests_previous_version` in need to fake 5.0 as 4.7. The whole block should be removed in master.

Reply:
ok

Reply:
i think the fix should instead happen in the "get_latest_previous_version"
this feels like a hacky way to resolve this...

Reply:
No we are not dependent to 4.7 manifest. We only pretend 5.0 to be 4.7 for the sake of this bash sourcery. The version 4.7 is used for calculation of 4.6 to create Subscription using operatorhub.